File: app/src/main/java/com/igtools/downloader/adapter/HistoryAdapter.java
package com.igtools.downloader.adapter;

import android.content.Context;
import android.graphics.drawable.ColorDrawable;
import android.media.Image;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.igtools.downloader.R;

import java.util.ArrayList;
import java.util.List;

/**
 * @Author: desong
 * @Date: 2022/8/4
 */
public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.HistoryHolder> {
    List<String> thumbnails = new ArrayList<>();
    List<String> titles = new ArrayList<>();
    List<String> usernames = new ArrayList<>();
    List<String> avatars = new ArrayList<>();
    Context c;
    OnItemClickListener onItemClickListener;

    public HistoryAdapter(Context c) {

        this.c = c;
    }

    @NonNull
    @Override
    public HistoryHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(c).inflate(R.layout.item_history, parent, false);
        return new HistoryHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull HistoryHolder holder, int position) {

        Glide.with(c).load(thumbnails.get(position))
                .placeholder(new ColorDrawable(ContextCompat.getColor(c, R.color.gray_1)))
                .into(holder.thumbnail);

        holder.caption.setText(titles.get(position));
        holder.itemView.setOnClickListener(v -> onItemClickListener.onClick(position));

        Glide.with(c)
                .load(avatars.get(position))
                .circleCrop()
                .placeholder(new ColorDrawable(ContextCompat.getColor(c, R.color.gray_1)))
                .into(holder.avatar);
        holder.username.setText(usernames.get(position));

    }

    @Override
    public int getItemCount() {
        return thumbnails.size();
    }

    public void setDatas(List<String> thumbnails, List<String> titles, List<String> usernames, List<String> avatars) {
        this.thumbnails = thumbnails;
        this.titles = titles;
        this.usernames = usernames;
        this.avatars = avatars;
        notifyDataSetChanged();
    }

    class HistoryHolder extends RecyclerView.ViewHolder {

        ImageView thumbnail;
        TextView caption;
        ImageView avatar;
        TextView username;

        public HistoryHolder(@NonNull View itemView) {
            super(itemView);

            thumbnail = itemView.findViewById(R.id.img_thumbnail);
            caption = itemView.findViewById(R.id.tv_caption);
            avatar = itemView.findViewById(R.id.img_avatar);
            username = itemView.findViewById(R.id.tv_username);
        }
    }

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }

    public interface OnItemClickListener {

        void onClick(int position);

    }
}

File: app/src/main/java/com/igtools/igdownloader/adapter/HistoryAdapter.kt
package com.igtools.igdownloader.adapter

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import com.igtools.igdownloader.adapter.HistoryAdapter.HistoryHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import com.igtools.igdownloader.R
import com.bumptech.glide.Glide
import android.graphics.drawable.ColorDrawable
import android.net.Uri
import android.view.View
import android.widget.ImageView
import androidx.core.content.ContextCompat
import android.widget.TextView
import com.igtools.igdownloader.models.MediaModel
import java.io.File
import java.util.ArrayList

/**
 * @Author: desong
 * @Date: 2022/8/4
 */
class HistoryAdapter(var c: Context) : RecyclerView.Adapter<HistoryHolder>() {

    var medias: List<MediaModel> = ArrayList()
    var onItemClickListener: OnItemClickListener? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HistoryHolder {
        val v = LayoutInflater.from(c).inflate(R.layout.item_history, parent, false)
        return HistoryHolder(v)
    }

    override fun onBindViewHolder(holder: HistoryHolder, position: Int) {
        Glide.with(c).load(medias[position].thumbnailUrl)
            .placeholder(ColorDrawable(ContextCompat.getColor(c, R.color.gray_1)))
            .into(holder.thumbnail)
        holder.caption.text = medias[position].captionText
        holder.itemView.setOnClickListener { v: View? -> onItemClickListener!!.onClick(position) }
        Glide.with(c)
            .load(medias[position].profilePicUrl)
            .circleCrop()
            .placeholder(ColorDrawable(ContextCompat.getColor(c, R.color.gray_1)))
            .into(holder.avatar)
        holder.username.text = medias[position].username
    }

    override fun getItemCount(): Int {
        return medias.size
    }

    fun setDatas(medias: ArrayList<MediaModel>) {
        this.medias = medias
        notifyDataSetChanged()
    }

    inner class HistoryHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var thumbnail: ImageView = itemView.findViewById(R.id.img_thumbnail)
        var caption: TextView = itemView.findViewById(R.id.tv_caption)
        var avatar: ImageView = itemView.findViewById(R.id.img_avatar)
        var username: TextView = itemView.findViewById(R.id.tv_username)

    }


    interface OnItemClickListener {
        fun onClick(position: Int)
    }
}